packages/webdriverio/src/utils/interception/devtools.js

```diff
@@ -29,11 +29,10 @@ export default class DevtoolsInterception extends Interception {
                  * match filter options
                  */
                 if (
-                    /**
-                     * HTTP method
-                     */
-                    (mock.filterOptions.method && mock.filterOptions.method.toLowerCase() !== request.method.toLowerCase()) ||
-                    (mock.filterOptions.headers && containsHeaderObject(responseHeaders, mock.filterOptions.headers))
+                    filterMethod(request.method, mock.filterOptions.method) ||
+                    filterHeaders(request.headers, mock.filterOptions.requestHeaders) ||
```

Comment:
I guess I should update `expect().toBeRequestedWith` to accept `requestHeaders` and `responseHeaders ` instead of just `headers`.
I was also thinking about merging headers like this `{ ...requestHeaders, ... responseHeaders }` but I'm afraid it may cause some unexpected issues.
@christian-bromann what do you think?

Reply:
> I guess I should update expect().toBeRequestedWith to accept requestHeaders and responseHeaders instead of just headers.

Sounds good to me.

> I was also thinking about merging headers like this { ...requestHeaders, ... responseHeaders } but I'm afraid it may cause some unexpected issues.

Why merging?

Reply:
https://github.com/webdriverio/expect-webdriverio/issues/205